mewarpx/tests/test_assemblies.py

```diff
@@ -164,3 +164,87 @@ def test_embedded_rectangle():
     ))
 
     assert np.allclose(phi, ref_phi, 0.001)
+
+
+def test_two_embedded_cylinders():
+    name = "two_embedded_cylinders_solve"
+    dim = 2
+
+    # Initialize and import only when we know dimension
+    mwxutil.init_libwarpx(ndim=dim, rz=False)
+    from mewarpx import assemblies
+    from mewarpx.setups_store import diode_setup
+    from mewarpx.utils_store import testing_util
+    from mewarpx.mwxrun import mwxrun
+
+    # Include a random run number to allow parallel runs to not collide. Using
+    # python randint prevents collisions due to numpy rseed below
+    testing_util.initialize_testingdir(name)
+
+    # Initialize each run with consistent, randomly-chosen, rseed. Use a random
+    # seed instead for initial dataframe generation.
+    # np.random.seed()
+    np.random.seed(42578825)
+
+    # Specific numbers match older run for consistency
+    D_CA = 1  # m
+    NX = 64
+    NZ = 64
+    run = diode_setup.DiodeRun_V1(
+        dim=dim,
+        #V_ANODE_CATHODE=VOLTAGE,
+        D_CA=D_CA,
+        NX=NX,
+        NZ=NZ,
+        # This gives equal spacing in x & z
+        PERIOD=D_CA * NX / NZ,
+        DT=1e-6,
+        TOTAL_TIMESTEPS=1,
+        DIAG_STEPS=1,
+        FIELD_DIAG_DATA_LIST=['phi'],
+    )
+    # Only the functions we change from defaults are listed here
+    run.setup_run(
+        init_conductors=False,
+        init_scraper=False,
+        init_electrons=False,
+        init_solver=False,
+        init_injectors=False,
+        init_field_diag=True,
+        init_simcontrol=True,
+        init_simulation=False
+    )
+
+    # Install the embedded boundaries
+    cylinder1 = assemblies.Cylinder(
+        center_x=-0.25, center_z=0.5, radius=0.1,
+        V=-2.0, T=300, WF=4.7, name="Cylinder1"
+    )
+    cylinder2 = assemblies.Cylinder(
+        center_x=0.25, center_z=0.5, radius=0.1,
+        V=5.0, T=300, WF=4.7, name="Cylinder2"
+    )
+
+    # Initialize solver
+    run.init_solver()
+    run.init_conductors()
+
+    # Initialize the simulation
+    run.init_simulation()
+    run.init_warpx()
+
+    # Run the main WARP loop
+    while run.control.check_criteria():
+        mwxrun.simulation.step()
+
+    #######################################################################
+    # Check phi results against reference data                            #
+    #######################################################################
+
+    phi = mwxrun.get_gathered_phi_grid(include_ghosts=False)
+    # np.save('two_embedded_cylinders_phi.npy', phi)
+    ref_phi = np.load(os.path.join(
+        testing_util.test_dir, 'embedded_boundary',
+        'two_embedded_cylinders_phi.npy'
+    ))
+    assert np.allclose(phi, ref_phi, 0.001)
```

Comment:
```suggestion
    assert np.allclose(phi, ref_phi, rtol=0.001)
```
Just saves people from looking up whether rtol or atol comes first/clearer code.

Reply:
There are a few more places in the same file where the same change is applicable, so I'll just add it in along with the next commit.